Tests/SonarScanner.MSBuild.PreProcessor.Test/PreprocessorObjectFactoryTests.cs

```diff
@@ -29,77 +31,107 @@ namespace SonarScanner.MSBuild.PreProcessor.Test
 
-        #region Tests
+        [TestInitialize]
+        public void TestInitialize() =>
+            logger = new TestLogger();
 
         [TestMethod]
         public void Factory_ThrowsOnInvalidInput()
         {
-            // Arrange
-            var logger = new TestLogger();
-            IPreprocessorObjectFactory testSubject = new PreprocessorObjectFactory(logger);
-
-            // 1. Ctor
-            Action act1 = () => new PreprocessorObjectFactory(null);
-            act1.Should().ThrowExactly<ArgumentNullException>();
+            Action ctor = () => new PreprocessorObjectFactory(null);
+            ctor.Should().ThrowExactly<ArgumentNullException>();
 
-            // 1. CreateSonarQubeServer method
-            Action act2 = () => testSubject.CreateSonarQubeServer(null);
-            act2.Should().ThrowExactly<ArgumentNullException>();
+            var sut = new PreprocessorObjectFactory(logger);
+            Action callCreateSonarQubeServer = () => sut.CreateSonarQubeServer(null);
+            callCreateSonarQubeServer.Should().ThrowExactly<ArgumentNullException>();
         }
 
         [TestMethod]
         public void Factory_ValidCallSequence_ValidObjectReturned()
         {
-            // Arrange
-            var logger = new TestLogger();
             var validArgs = CreateValidArguments();
-            IPreprocessorObjectFactory testSubject = new PreprocessorObjectFactory(logger);
+            var sut = new PreprocessorObjectFactory(logger);
 
-            // 1. Create the SonarQube server...
-            object actual = testSubject.CreateSonarQubeServer(validArgs);
+            object actual = sut.CreateSonarQubeServer(validArgs);
             actual.Should().NotBeNull();
 
-            // 2. Now create the targets provider
-            actual = testSubject.CreateTargetInstaller();
+            actual = sut.CreateTargetInstaller();
             actual.Should().NotBeNull();
 
-            // 3. Now create the analyzer provider
-            actual = testSubject.CreateRoslynAnalyzerProvider();
+            actual = sut.CreateRoslynAnalyzerProvider();
             actual.Should().NotBeNull();
         }
 
         [TestMethod]
         public void Factory_InvalidCallSequence_Fails()
         {
-            // Arrange
-            var logger = new TestLogger();
-            IPreprocessorObjectFactory testSubject = new PreprocessorObjectFactory(logger);
+            var sut = new PreprocessorObjectFactory(logger);
 
-            // 2. Act and assert
-            Action act = () => testSubject.CreateRoslynAnalyzerProvider();
+            Action act = () => sut.CreateRoslynAnalyzerProvider();
             act.Should().ThrowExactly<InvalidOperationException>();
         }
 
-        #endregion Tests
+        [TestMethod]
+        public void CreateHttpClient_Authorization()
+        {
+            GetHeader(PreprocessorObjectFactory.CreateHttpClient(null, null, null, null), HttpRequestHeader.Authorization).Should().BeNull();
```

Comment:
Let's extract `GetHeader`+`PreprocessorObjectFactory` to a local method that will take `userName` and `password` arguments. Something like `AuthorizationHeader(..., ..).Should().Be...`

Reply:
I mentioned `local`, but it works either way